edx_sysadmin/api/test_permissions.py

```diff
@@ -0,0 +1,70 @@
+"""
+Tests for Permissions
+"""
+
+from unittest.mock import patch
+from django.conf import settings
+from django.test import TestCase, override_settings
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+from rest_framework.response import Response
+
+
+class GithubWebhookPermissionTestCase(TestCase):
+    """
+    Test Case for GithubWebhookPermission permission
+    """
+
+    SYSADMIN_GITHUB_WEBHOOK_KEY = "nuiVypAArY7lFDgMdyC5kwutDGQdDc6rXljuIcI5iBttpPebui"
+    VALID_SIGNATURE = "313aa3f017c815f6677f66d9acb87cee1adc0a3ef2998b7add789aab0632a0e6"
+
+    def setUp(self):
+        super().setUp()
+        self.client = APIClient()
+
+    def test_GithubWebhookPermission_without_key(self):
+        """Test GithubWebhookPermission without signature key"""
+
+        response = self.client.post(reverse("sysadmin:api:git-reload"))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @override_settings(SYSADMIN_GITHUB_WEBHOOK_KEY=SYSADMIN_GITHUB_WEBHOOK_KEY)
+    @patch(
```

Comment:
The reason behind patching is that we don't want to test the permission class itself. If the permissions passed then the control will be given to `post` method and if that post method raised an error we won't be able to know from where it came. That's why we are asserting on  mocked `post` method to check the permissions only in these tests.

The tests for views should be separately added because it has a lot of scenarios to be covered. And to add them, I believe, firstly we need to finalize this current PR.

Reply:
@HamzaIbnFarooq I think we should add tests for view in this PR because having tests and implementation in same PR makes more sense unless we have very large changeset in the PR.

Reply:
@HamzaIbnFarooq about patching `GitReloadAPIView.post` what if we don't patch? In case of permission pass we'll get a 400 status code which is fine. I mean if we are not geting 403 then it means our permission checks are passed.

Reply:
@ziafazal yes that makes sense, let me add that accordingly and ping you after wards.